resources/js/frontend/wc-payplug-apple_pay-blocks.js

```diff
@@ -218,25 +218,25 @@ const ExpressApplePay = {
 	edit: <ExpressContent/>,
 	canMakePayment: (data) => {
 
-		settings.payplug_carriers.forEach(function(item){
-
-			if (item.identifier === data.selectedShippingMethods[0].split(":")[0]){
-				item.selected = true;
-			}
-
-		})
-
-		settings.payplug_apple_pay_shipping_required = data.cartNeedsShipping;
-		settings.total_amount = data.cartTotals.total_price;
-
 		if (!settings?.is_cart) {
 			return false;
 		}
 
+		settings.payplug_apple_pay_shipping_required = data.cartNeedsShipping;
+		settings.total_amount = data.cartTotals.total_price;
+
 		if (!data.cartNeedsShipping) {
 			return true;
 		}
 
+		settings.payplug_carriers.forEach(function(item){
+
+			if (item.identifier === data.selectedShippingMethods[0].split(":")[0]){
```

Comment:
Maybe data.selectedShippingMethods[0].split(":")[0] can be defined as a contant?